aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -393,7 +398,20 @@ inline void bgemm_internal_cublaslt(CUDABLAS_BGEMM_ARGTYPES(Dtype)) {
   preference.setAttribute(CUBLASLT_MATMUL_PREF_MIN_ALIGNMENT_C_BYTES, c_alignment);
 #endif
 
-  auto workspace = at::empty(static_cast<int64_t>(workspaceSize), at::TensorOptions().dtype(at::kByte).device(at::kCUDA));
+#ifdef USE_ROCM
+  auto& allocator = *::c10::cuda::CUDACachingAllocator::get();
+  auto workspace = allocator.allocate(workspaceSize);
+  auto workspace_ptr = workspace.mutable_get();
+  TORCH_CHECK(workspace_ptr != nullptr, "OOM trying to allocate workspace for cublaslt");
+#else
+  cublasHandle_t handle = at::cuda::getCurrentCUDABlasHandle();
```

Comment:
1) Do we need to use divergent code paths for ROCm? It seems in cublas workspace code there's no divergence
2) This is called 3 times, can you abstract getting a workspace into a function to avoid this copy pasta?
3) You already expect cublas workspace to exist `TORCH_CHECK(workspace_it != cublas_handle_stream_to_workspace().end())`, I don't see how it's guaranteed

Reply:
1. right now I'm a bit hesitant to also change it for ROCM as their Lt equivalent uses a different default workspace size from the legacy interface / workspace size: https://github.com/pytorch/pytorch/pull/139300
2. Sure, moved into function
3. It's expected as the "get handle" function should do the initial allocation if it is not allocated already---not guaranteed due to e.g., OOM